test/collection.js

```diff
@@ -1431,6 +1431,25 @@
     equal(c.models.length, 1);
   });
 
+  test('Do not allow duplicate models (with parsed id) to be `add`ed or `set`', function() {
+    var Stooge = Backbone.Model.extend({
+      idAttribute: '_id',
+      parse: function(a) {
+        var attrs = {};
+        attrs._id = a.__id;
+        return attrs;
+      }
+    });
+    var StoogeCollection = Backbone.Collection.extend({model: Stooge});
+    var c = new StoogeCollection();
+    c.add([{__id: 1}, {__id: 1}], {parse: true});
```

Comment:
Can we clean up these tests to use, e.g. a nested resource instead of a weird `id` example? This is kinda hard to follow.